### PR TITLE
Do zero checking to avoid unnecessary work in addToExpression

### PR DESCRIPTION
**Pull request by joehuchette**
ref #535

### CODE REVIEW

**Comment by IainNZ** [comment]
Nice. Just needs 0.3

**Comment by joehuchette** [comment]
@IainNZ is this the same (terrifying) Travis error you were seeing?

**Comment by IainNZ** [comment]
Sure is

**Comment by IainNZ** [comment]
I kinda love that AppVeyor is finally the one passing and the others are failing

File: src/v0.3/parseExpr_0.3.jl [hunk 1]
```diff
@@ -39,15 +41,19 @@ function addToExpression(aff::GenericAffExpr,c::Number,x::Number)
 end
 
 function addToExpression{C,V}(aff::GenericAffExpr{C,V}, c::Number, x::V)
```
Comment (mlubin) [inline src/v0.3/parseExpr_0.3.jl]: Is it me or is this method identical to the one above on line 31?

Comment (joehuchette) [inline src/v0.3/parseExpr_0.3.jl]: Hah

Comment (joehuchette) [inline src/v0.3/parseExpr_0.3.jl]: No, they're the same

**Comment by joehuchette** [comment]
The Travis failure looks like a recompilation issue with DataStructures; I can reproduce locally.

**Comment by joehuchette** [comment]
Actually I think DataStructures was just in a broken state, I officially have no idea what's going on now.

**Comment by mlubin** [comment]
Report upstream? I can reproduce with ``using JuMP``.

**Pull request merged**